src/pwa/components/PaginationWaypoint.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import { string, number, func, node } from 'prop-types';
+import { inject } from 'mobx-react';
+import Waypoint from 'react-waypoint';
+
+class PaginationWaypoint extends Component {
+  static counter = 1;
+
+  static lastPageEntered = 0;
+
+  static lastHtmlTree = {};
+
+  static propTypes = {
+    page: number.isRequired,
+    title: string.isRequired,
+    url: string.isRequired,
+    sendPageView: func.isRequired,
+    children: node.isRequired,
+  };
+
+  constructor() {
+    super();
+
+    this.getPageviewPayload = this.getPageviewPayload.bind(this);
```

Comment:
Not that it matters but binding is not needed if you call it only from `sendPageView(this.getPageviewPayload());`. Binding is only useful when you are passing the reference of the function, like in `onEnter={this.handleOnEnter}`.